src/PostgREST/App.hs

```diff
@@ -391,7 +395,10 @@ handleSingleUpsert identifier context@(RequestContext _ _ ApiRequest{..} _) = do
       response HTTP.status204 [] mempty
 
 handleDelete :: QualifiedIdentifier -> RequestContext -> DbHandler Wai.Response
-handleDelete identifier context@(RequestContext _ _ ApiRequest{..} _) = do
+handleDelete identifier context@(RequestContext _ ctxDbStructure ApiRequest{..} _) = do
+  when (iTopLevelRange /= RangeQuery.allRange && findIfView identifier (dbTables ctxDbStructure)) $
+    throwError $ Error.NotImplemented "limit/offset is not implemented for views"
+
```

Comment:
Having this in `App.hs` really seems the wrong place. This should go into `ApiRequest.hs`, I think - because this module is concerned with parsing the user's request and matching it up with the schema cache.

Reply:
>This should go into ApiRequest.hs, I think - because this module is concerned with parsing the user's request and matching it up with the schema cache.

Hmm, I never thought of ApiRequest like that, I mean the only reason why the schema cache is used there is because of the function overloading which needs the Proc info and that is something that I thought be better moved to a different module.

If you check `handleSingleUpsert` it has a similar logic:

https://github.com/PostgREST/postgrest/blob/a3b0908548dba9f2c6df977c7076f46ec9b19ba8/src/PostgREST/App.hs#L373-L376

So to be consistent I thought `App.hs` was the better place for this logic.

Reply:
Interesting. The `handleSingleUpsert` doesn't even depend on the schema cache, so it's even more out of place. It's an invalid request, so that should really go into `AppRequest`.